packages/tool_util/test-requirements.txt

```diff
@@ -1,3 +1,4 @@
-beaker
+beaker==1.11.0 ; python_version >= "3.7" and python_version < "3.12"
+sqlalchemy==1.4.46 ; python_version >= "3.7" and python_version < "3.12"
```

Comment:
The package requirements are ideally unpinned. It seems that 1.12.1 fixes your issue, so I think we should remove the pin here (ideally for sqlalchemy too, or if 2.0 fails, to pin it to `sqlalchemy<=2`)

Reply:
There's typically a distinction being made between libraries and applications. Galaxy as an application pins all dependencies hard (i.e to specific, tested versions, see https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/dependencies/pinned-requirements.txt), while libraries should be pinned lightly, so that as the author of an application you have a good chance of being able to pick a set of compatible requirements.

Reply:
In addition to Marius' comment, the central list of compatible "light" pinnings for Galaxy Python dependencies is in `pyproject.toml` , and the various `*requirements.txt` files inside `packages/` should mirror them, see my other comment.